A37-xds-aggregate-and-logical-dns-clusters.md

```diff
@@ -0,0 +1,325 @@
+A37: xDS Aggregate and Logical DNS Clusters
+----
+* Author(s): Mark D. Roth (roth@google.com)
+* Approver: ejona86
+* Status: {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2021-01-26
+* Discussion at: https://groups.google.com/g/grpc-io/c/bZ7wLIdNCVY
+
+## Abstract
+
+gRPC will add support for the xDS [aggregate cluster extension](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/aggregate_cluster.html)
+and [logical DNS cluster type](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#logical-dns).
+These two features, in combination, will allow configuring a fallback
+from EDS to DNS for a given cluster, which may ease transition to xDS in
+some environments.
+
+## Background
+
+Currently, gRPC supports only the [EDS cluster type](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/upstream/service_discovery#endpoint-discovery-service-eds).
+However, in some environments, endpoint discovery is instead done via
+DNS.  It is desirable to be able to support such environments, and it is
+desirable to allow them to migrate from DNS to EDS in a way that allows
+them to fall back to DNS if the new EDS service fails.
+
+It should be noted that the intended xDS aggregate cluster behavior has
+recently been fleshed out, and Envoy's implementation of this behavior
+will be changing to meet the new spec.  gRPC's implementation will meet
+this new spec from the start.  For details, see
+https://github.com/envoyproxy/envoy/issues/13134.
+
+### Related Proposals: 
+
```

Comment:
It would be great to link the public [xDS Aggregate Cluster Behavior](https://docs.google.com/document/d/17fc0rAktQpc5oMkerOLHSTnXwX1xWAqW7vZ6_LL47NY/edit#heading=h.4xxf5tk5hhr0) doc as well.

Reply:
That's linked to from the Envoy github issue above.